### ISSUE TITLE
Registering buffers in Components

### ISSUE THREAD

**Issue by prabhuteja12** [awslabs/Renate#236]
The [Component](https://github.com/awslabs/Renate/blob/dev/src/renate/updaters/learner_components/component.py#L74) class implements a [`_register_parameters` ](https://github.com/awslabs/Renate/blob/dev/src/renate/updaters/learner_components/losses.py#L34) method that registers the loss' constructor arguments. 

This currently works without glitches where Renate manages the `state_dict` of Learner, Model and others. However, once the move to relying on Lightning for all checkpointing is complete (in #218) this fails as registering the constructor args (say `_weight`) returns it in the state_dict, and causes an error when restoring the model from checkpoint. The 'fix' for now is going to by changing [`RenateModule`](https://github.com/awslabs/Renate/blob/dev/src/renate/models/renate_module.py#L77)'s `load_state_dict` to use `strict=False`. 

This issue should help cleanup the components code too.
